Simplified/Reader2/Bookmarks/NYPLReadiumBookmark+R2.swift

```diff
@@ -17,28 +17,34 @@ extension NYPLReadiumBookmark {
   /// for this conversion: only what's strictly necessary to be able to point
   /// at the same location inside the `Publication`.
   ///
-  /// - Complexity: O(*n*) where *n* is the length of internal `Publication`
-  /// data structures, such as list of chapters, resources, links.
+  /// - Complexity: O(*n*) where *n* is the length of internal
+  /// `Publication.readingOrder` data structure.
   ///
   /// - Parameter publication: The R2 publication object where the bookmark is
   /// located.
 
   /// - Returns: An object with R2 location information pointing at the same
   /// position the bookmark model is pointing to.
   func convertToR2(from publication: Publication) -> NYPLBookmarkR2Location? {
-    guard let link = publication.link(withIDref: self.idref) else {
+    let href: String!
```

Comment:
I think we can avoid implicit unwrapping here

Reply:
this implicit unwrapping is 100% safe in the code below, but I couldn't find a way that is more readable. I'm open to ideas though, do you have something specific in mind?

Reply:
this implicit unwrapping is 100% safe in the code below, but I couldn't find a way that is more readable. I'm open to ideas though, do you have something specific in mind?

Reply:
fixed and rebased